container/hash.bzl

```diff
@@ -0,0 +1,40 @@
+# Copyright 2017 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Functions for producing the hash of an artifact."""
+
+def sha256(ctx, artifact, execution_requirements = None):
```

Comment:
Why should we not just upstream these changes to https://github.com/bazelbuild/bazel/blob/master/tools/build_defs/hash/hash.bzl?

Reply:
Simply put, we should. I'll put up a PR for that as well.

Reply:
https://github.com/bazelbuild/bazel/pull/10405